[python] remove deprecated cellxgene.experimental.ml modules

mlin opened the pull request:
Planned for ~June 2025

mlin commented:
@ivirshup got a nudge from @dsadgat that now could be a good time to merge this in as planned, WDYT?

(also deals with some CI bitrot- new linter versions etc.)

ivirshup commented on api/python/cellxgene_census/pyproject.toml:
```diff
@@ -40,9 +40,6 @@ dependencies= [
 experimental = [
-    "torch",
-    "torchdata~=0.7,<0.10",  # FIXME: delete when the deprecated pytorch loader API prototype is fully graduated to TileDB-SOMA-ML
-    "scikit-learn>=1.2",
```
While we're at it, should we remove `torchdata` from the intersphinx config?

https://github.com/chanzuckerberg/cellxgene-census/blob/99583e2d7deed4cde75a3d7915bb3396d7cacb81/docs/conf.py#L52-L54

I think `scikit-learn` and `torch` are still needed in intersphinx for the data loader notebook (which I think we're fine to keep overall since it uses the tiledb loader).

Though also intersphinx does not seem to be working in that notebook....

ivirshup reviewed (commented):
Thanks for pinging me about this!

I think you've got everything in the code, but I think there's more to remove in the docs. E.g. the objects linked in `docs/python-api.rst` ([docsite](https://chanzuckerberg.github.io/cellxgene-census/python-api.html#experimental-machine-learning )) and the "new to the census, pytorch loaders!" announcement from the landing page at [`docs/cellxgene_census_docsite_landing.md`](https://github.com/chanzuckerberg/cellxgene-census/blob/main/docs/cellxgene_census_docsite_landing.md).

I am a little unsure what to do about this article: https://chanzuckerberg.github.io/cellxgene-census/articles/2024/20240709-pytorch.html. Maybe needs a note pointing to the current data loader notebook?

mlin commented:
@ivirshup Well spotted- yea I removed the TOC section, took down the landing page notice (and also the "Coming Soon" section which seems defunct), took out torchdata from intersphinx conf.py, and added a notice to that article.
